File: leetcode/problems/find-the-index-of-the-first-occurrence-in-a-string/v1.java

class Solution {
    public int strStr(String haystack, String needle) {
        int i = 0;
        while (i < haystack.length()) {
            int j = 0;
            while (j < needle.length()) {
                if (haystack.charAt(i + j)!= needle.charAt(j)) break;
                j++;
                if (j == needle.length()) {
                    return i;
                }
                if (i + j == haystack.length()) {
                    return -1;
                }
            }
            i++;
        }
        return -1;
    }
}

public class v1 {
    public static void main(String args[]) {
        Solution solution = new Solution();
        int ans = solution.strStr("subsubsub", "sub");
        System.out.println(ans);
    }
}

File: leetcode/problems/find-the-index-of-the-first-occurrence-in-a-string/v1.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        n = len(haystack)
        m = len(needle)
        i = 0
        while i < n:
            j = 0
            while haystack[i + j] == needle[j]:
                j += 1
                if j == m:
                    return i
                if i + j == n:
                    return -1
            i += 1
        return -1


solution = Solution()
ans = solution.strStr("sadbutsad", "sad")
print(ans)
ans = solution.strStr("leetcode", "leeto")
print(ans)
